pkg/bpf/map_linux.go

```diff
@@ -633,6 +644,35 @@ func (m *Map) DumpWithCallback(cb DumpCallback) error {
 	return i.Err()
 }
 
+type DumpPerCPUCallback func(key MapKey, values any)
+
+func (m *Map) DumpPerCPUWithCallback(cb DumpPerCPUCallback) error {
+	if cb == nil {
+		return errors.New("empty callback")
+	}
+
+	if err := m.Open(); err != nil {
```

Comment:
Not sure of implications here but the map is open and never closed and the path is not checked? 
It looks like the call site also manages existence/open/close on the map?
Do we want to assume call site is responsible for these checks if existence is required?
Would it be best to let this method handle it instead and perhaps DumpPerCPUWithCBIfExists?

Reply:
@gentoo-root Thank you for the clarification! Are there implications in opening the map twice? The current pattern seems off as the call site opens the resource and is responsible to close it but the underlying call re-open the resource and assumes the call site is responsible for closing it?? 
This might be ok on a short lived call but it seems to be an anti-pattern.

Reply:
@gentoo-root Thanks for the clarifications! I think at the very least we should add docs to that effect on all the public APIs to ensure the call site is aware and does the right thing with that resource. 
And perhaps a TODO might be appropriate here so we can revisit for a code debt pass